### PR TITLE
Add info.olasagasti.revelation

### PR DESCRIPTION
**Pull request by mikelolasagasti**
Signed-off-by: Mikel Olasagasti Uranga <mikel@olasagasti.info>

### CODE REVIEW

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

File: info.olasagasti.revelation.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
+    "app-id": "info.olasagasti.revelation",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "rename-desktop-file" : "revelation.desktop",
+    "rename-icon" : "revelation",
+    "rename-icon" : "revelation-locked",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:8]: It can be specified only once.

File: info.olasagasti.revelation.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+    "app-id": "info.olasagasti.revelation",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "rename-desktop-file" : "revelation.desktop",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:6]: Ideally all these renames should happen upstream.

File: info.olasagasti.revelation.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "rename-icon" : "revelation",
+    "rename-icon" : "revelation-locked",
+    "command" : "revelation",
+    "finish-args" : [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:16]:
```suggestion
```

File: info.olasagasti.revelation.json [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        "--filesystem=xdg-run/gvfs",
+        "--filesystem=/tmp",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:18]: What do you need `/tmp/access` for?

File: info.olasagasti.revelation.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        "--filesystem=home",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:19]: Does it need full home access?

File: info.olasagasti.revelation.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:20]:
```suggestion
```

File: info.olasagasti.revelation.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "rename-desktop-file" : "revelation.desktop",
+    "rename-icon" : "revelation",
+    "rename-icon" : "revelation-locked",
+    "command" : "revelation",
+    "finish-args" : [
+        "--share=ipc",
+        "--share=network",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:12]: Does it need network access?

**Comment by barthalion** [comment]
I suspect appdata file should be either in `/app/share/appdata` or in `/app/share/metainfo` but with `metainfo.xml` extension.

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

File: info.olasagasti.revelation.json [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "revelation",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/gvfs",
+        "--filesystem=home",
+        "--talk-name=ca.desrt.dconf",
```
Comment (bilelmoussaoui) [inline info.olasagasti.revelation.json:13]:
```suggestion
```

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

File: info.olasagasti.revelation.json (hunk 5 above)
Comment (barthalion) [reply info.olasagasti.revelation.json:12]: Okay, if this is about opening/saving files, `GtkFileChooserNative` should have access everywhere without poking holes in the sandbox. `VFS` permissions could be dropped then too.

**Comment by barthalion** [comment]
The app uses stock icon for the desktop launcher, or at least it displays that way here.

image

**Comment by nedrichards** [comment]
bot, build info.olasagasti.revelation

**Comment by nedrichards** [comment]
There's a bunch of missing menu bar icons as well:
File->Change Password, Lock, Import and Export
View->Password Generator, Password Checker

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
> There's a bunch of missing menu bar icons as well:
> File->Change Password, Lock, Import and Export
> View->Password Generator, Password Checker

This icons should be present now

File: info.olasagasti.revelation.json (hunk 5 above)
Comment (mikelolasagasti) [reply info.olasagasti.revelation.json:12]: I'll need some time to port GtkFileChooserDialog to GtkFileChooserNative. Is this a hard requirement or could be changed later?

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

File: info.olasagasti.revelation.json [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            "name" : "revelation",
+            "buildsystem" : "autotools",
+            "config-opts": [
+                "--disable-mime-update"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/mikelolasagasti/revelation.git",
+                    "branch" : "flathub-change"
```
Comment (bilelmoussaoui) [inline info.olasagasti.revelation.json:56]: This needs to be a tagged release or a specific commit for builds reproducibility.

Comment (mikelolasagasti) [reply info.olasagasti.revelation.json:56]: I was using a branch to test changes required to build the flatpak on flathub that will merged and released. I was planning to do it after the review was complete, would it be mandatory to do it before/now?

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
bot, build info.olasagasti.revelation

**Comment by mikelolasagasti** [comment]
Manifest changed to new release with changes required by flathub.

The only missing part would be the /home permission, that requires the change from `Gtk.FileChooserDialog` to `Gtk.FileChooserNative`. Would it be possible to accept the Manifest as is now and I'll take care of removing full home access once app is migrated to `Gtk.FileChooserNative`?

File: info.olasagasti.revelation.json [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        "--socket=wayland",
+        "--filesystem=xdg-run/gvfs",
+        "--filesystem=home",
+        "--talk-name=org.gtk.vfs",
+        "--talk-name=org.gtk.vfs.*",
+        "--talk-name=org.gnome.ScreenSaver"
+    ],
+    "modules" : [
+        {
+            "name" : "libpwquality",
```
Comment (barthalion) [inline info.olasagasti.revelation.json:19]: Would be nice to add `"cleanup": ["*.a"]` but the file is small so it can be done later.

**Pull request closed without merging**